### PR TITLE
Update Coiled links

### PR DESCRIPTION
**Pull request by scharlottej13**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

Small PR to 1) make sure Coiled-related links still work/make sense and 2) to add UTM codes to links.

cc @jacobtomlinson @jrbourbeau

### CODE REVIEW

File: docs/source/best-practices.rst [hunk 1]
```diff
@@ -369,5 +369,3 @@ Compute related results with shared computations in a single :func:`dask.compute
 
 This allows Dask to compute the shared parts of the computation (like the
 ``dd.read_csv`` call above) only once, rather than once per ``compute`` call.
-For more guidance on when to call ``compute`` when working with Dask DataFrames
-or Arrays see `this blog post <https://www.coiled.io/blog/dask-compute>`_.
```
Comment (scharlottej13) [inline docs/source/best-practices.rst:373]: Coiled removed this post. After reading the archived version again, I don't think it adds much. I considered linking to https://distributed.dask.org/en/stable/manage-computation.html#dask-collections-to-concrete-values, but was worried it'd be throwing new folks into the deep end.

**Review by jrbourbeau — APPROVED** [review]
Thanks @scharlottej13. This looks reasonable to me.

File: docs/source/support.rst [hunk 2]
```diff
@@ -90,5 +90,5 @@ In addition to the previous options, paid support is available from the
 following organizations (listed in alphabetical order):
 
 -   Anaconda: `<https://www.anaconda.com/products/professional-services>`_
--   Coiled: `<https://coiled.io>`_
+-   Coiled: `<https://coiled.io?utm_source=dask-docs&utm_medium=support>`_
```
Comment (jrbourbeau) [inline docs/source/support.rst:93]: Actually the link here looks off now 

Screenshot 2023-05-17 at 4 59 09 PM

Let's just have this list be a company names that are hyperlinked

Comment (scharlottej13) [reply docs/source/support.rst]: Ah thanks for catching this! Just pushed up a change

**Review by jacobtomlinson — APPROVED** [review]
This looks great thanks @scharlottej13

**Pull request merged**